### PR TITLE
Modified pipeline

### PR DESCRIPTION
**Pull request by harpercheng91**
1. batch normalization
- combine seurat integration report (integration_report.Rmd) and scran normalization report (batch_normalization.Rmd) into one script (batch_normalization.Rmd; the corresponding snakemake rule is do_batch_correction): visualize dim reduction of scran and seurat integration and batch correction.
- save seurat integrated and scran combined and corrected rds which are used in cell assign reports. We add cell type information to integrated objects instead of running integration on sce_cas to save time.
- combined and uncorrected objects are no longer saved because they are not used in downstream analyses. 
- helper function, seurat_integrate(), for making integrated objects from seurat integration is in utilities.R. Delete integration.R. 
- remove seurat integration rules (seurat_integration and integration_report) from snakemake.

2. summary_stats
- add PCA, tSNE and UMAP showing total number of detected reads on a log-scale to summary_stats_new.Rmd. 
- add titles and table of content for easy navigation.
- the current summary_stats.Rmd only accepts two samples. make this robust to any number of samples
make it robust to mouse and human sample
- only summary_stats.Rmd uses sce_raw and sce_qc. source make_sce and do_qc scripts and use these two objects directly instead of saving them

3. cellassign
- combined individual cellassign results with integrated results. The only rmd that remains is integration_report_cellassign.Rmd where we visualize dim reduction of individual samples, seurat integrated samples and scran combined and corrected samples. 
- the rule for running cell assign was deleted from snakemake. The only rule that remains is integrate_cell_assign_results (integration_report_cellassign.Rmd). run_cellassign.R is sourced in this rmd to assign cell type. Integrated objects from batch_correction output are loaded and cell type information is added. Here is what the rmd does:
   * load individual sce_clus
   * run cellassign 
   * load seurat_corrected and scran_corrected objects
   * append cell type information to integrated objects
   * dim reduction plots showing clusters and cell type on individual and integrated samples
- modified create_marker_matrix.R to let users specify the sample type (e.g., granulosa samples, ...) and it will create the corresponding marker gene matrix specific to your sample type. create_marker_matrix.R is sourced in run_cellassign.R, so we don't need to save marker gene matrix. 
- old scripts and rmd are moved to the 'archives/cellassign' folder.

4. cell cycle
- delete snakemake rule for running cell cycle assignment. Instead, source the script and run cyclone in rmd. We save two sce objects:
   * cell_cycle_label.rds: a column in rowData saying if a gene is cell cycle related. We can use this column to filter out cell cycle related genes in DGE analysis.
   * cell_cycle_corrected.rds: consider cell cycle as a batch effect and correct for it. We can use this sce directly in DGE analysis.
- in the rmd, we check if these two sces already exist before running cyclone. Because cyclone takes a long time to run, we don't want to repeat this process when rendering the report if we already have these objects.

5. clustering
we used to have clustering reports for individual samples and integrated samples. I think they might be redundant now that clustering for individual sample, seurat integrated, and scran combined and corrected sample can be visualized in cell assign reports.

6. archives folder:
- moved some unused scripts and rmds of cellassign and summary_stats to this folder.
- DBZ_paper, extract_info, and make_heatmaps seem to be project-specific. I moved them to archives as well since they are not relevant in terms of the general running of the pipeline.

### CODE REVIEW

**Comment by harpercheng91** [comment]
- created an option in process_qc_normalization where you can specify `save_qc` as "yes" if you want to save sce_qc.rds.
- made a new rule called scSorter_cellassign to show cell assignment with cellassign and scSorter. There are also expression level plots for a cell type of your choice.
- changed mesenchymal stem cell markers - detail is shown in the spreadsheet.
- some other trivial changes that fixed bugs
